src/Microsoft.Data.Sqlite/SqliteFactory.cs

```diff
@@ -0,0 +1,33 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Data.Common;
+
+namespace Microsoft.Data.Sqlite
+{
+    public class SqliteFactory : DbProviderFactory
+    {
+        static SqliteFactory()
+        {
+            Instance = new SqliteFactory();
+        }
+
+        private SqliteFactory()
+        {
+        }
+
+        public static SqliteFactory Instance { get; }
```

Comment:
It is old school magic :smile: BTW, I do agree things in .NET might by convention find the `Instance` field using reflection so if we are going to add it is better to follow the same pattern. @ajcvickers might have other insights.